test/runtime_test.cc

```diff
@@ -245,6 +247,40 @@ TEST_P(TestVM, Callback) {
   ASSERT_EQ(res.u32(), 100100); // 10000 (global) + 100(in callback)
 }
 
+TEST_P(TestVM, TerminateExecution) {
+  if (engine_ != "v8") {
+    return;
+  }
+  auto source = readTestWasmFile("callback.wasm");
+  ASSERT_TRUE(vm_->load(source, {}, {}));
+
+  TestContext context;
+  vm_->registerCallback(
+      "env", "callback", &callback,
+      &ConvertFunctionWordToUint32<decltype(callback), callback>::convertFunctionWordToUint32);
+
+  vm_->registerCallback(
+      "env", "callback2", &callback2,
+      &ConvertFunctionWordToUint32<decltype(callback2), callback2>::convertFunctionWordToUint32);
+
+  std::thread terminate([&]() {
+    std::this_thread::sleep_for(std::chrono::seconds(3));
+    vm_->terminateExecution();
+  });
+
+  ASSERT_TRUE(vm_->link(""));
+  WasmCallWord<1> run2;
+  vm_->getFunction("infinite_loop", &run2);
+  EXPECT_TRUE(run2 != nullptr);
+
+  run2(&context, Word{0});
```

Comment:
This should be named `infinite_loop` (after the function it calls), and it doesn't look that the extra parameter is used for anything.
```suggestion
  WasmCallVoid<0> infinite_loop;
  vm_->getFunction("infinite_loop", &infinite_loop);
  EXPECT_TRUE(infinite_loop != nullptr);

  infinite_loop(&context);
```